x/crosschain/keeper/abci.go

```diff
@@ -175,3 +185,55 @@ func CheckAndUpdateCctxGasPrice(
 
 	return gasPriceIncrease, additionalFees, nil
 }
+
+// CheckAndUpdateCctxGasRateBTC checks if the retry interval is reached and updates the gas rate if so
+// Zetacore only needs to update the gas rate in CCTX and fee bumping will be handled by zetaclient
+func CheckAndUpdateCctxGasRateBTC(
+	ctx sdk.Context,
+	k Keeper,
+	cctx types.CrossChainTx,
+	flags observertypes.GasPriceIncreaseFlags,
+) (math.Uint, math.Uint, error) {
+	// skip if gas price or gas limit is not set
+	if cctx.GetCurrentOutboundParam().GasPrice == "" || cctx.GetCurrentOutboundParam().CallOptions.GasLimit == 0 {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// skip if retry interval is not reached
+	lastUpdated := time.Unix(cctx.CctxStatus.LastUpdateTimestamp, 0)
+	if ctx.BlockTime().Before(lastUpdated.Add(flags.RetryInterval)) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// compute gas price increase
+	chainID := cctx.GetCurrentOutboundParam().ReceiverChainId
+	medianGasPrice, _, isFound := k.GetMedianGasValues(ctx, chainID)
+	if !isFound {
+		return math.ZeroUint(), math.ZeroUint(), cosmoserrors.Wrap(
+			types.ErrUnableToGetGasPrice,
+			fmt.Sprintf("cannot get gas price for chain %d", chainID),
+		)
+	}
```

Comment:
Can't this be put in common between BTC and EVM?

Reply:
This common logic is wrapped in `CheckAndUpdateCCTXGasPrice`